runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java

```diff
@@ -199,6 +200,36 @@ private XpWorldType worldSetToType(EnumSet<WorldType> types)
 		return xpType;
 	}
 
+	/**
+	 * Adds an overlay to the canvas for tracking a specific skill.
+	 *
+	 * @param skill the skill for which the overlay should be added
+	 */
+	void addOverlay(Skill skill)
+	{
+		removeOverlay(skill);
+		overlayManager.add(new XpInfoBoxOverlay(this, skill, skillIconManager.getSkillImage(skill)));
+	}
+
+	/**
+	 * Removes an overlay from the overlayManager if it's present.
+	 *
+	 * @param skill the skill for which the overlay should be removed.
+	 */
+	void removeOverlay(Skill skill)
+	{
+		overlayManager.removeIf(e -> e instanceof XpInfoBoxOverlay && ((XpInfoBoxOverlay) e).getSkill() == skill);
+	}
+
+	/**
+	 * Removes all instances of XpInfoBoxOverlay.
+	 * This is called when the plugin is reset to prevent the overlays from persisting.
+	 */
+	private void removeAllOverlays()
```

Comment:
this probably do not needs to be method as it is super simple and called only from 1 place.